catalogue_api/api/src/main/scala/uk/ac/wellcome/platform/api/finatra/exceptions/ElasticsearchResponseExceptionMapper.scala

```diff
@@ -56,9 +32,57 @@ class ElasticsearchResponseExceptionMapper @Inject()(
   //
   // When returning a 400 to the user, we wrap this error to avoid talking
   // about internal Elasticsearch concepts.
-  val resultSizePattern =
+  private val resultSizePattern =
     """Result window is too large, from \+ size must be less than or equal to: \[([0-9]+)\]""".r.unanchored
 
+  override def toResponse(request: Request,
+                          exception: ResponseException): Response = {
+    val result = toError(exception = exception)
+    val version = getVersion(request, s"$apiPrefix")
+
+    val errorResponse = ResultResponse(
+      context = buildContextUri(
+        apiScheme,
+        apiHost,
+        apiPrefix,
+        version,
+        apiContextSuffix),
+      result = result)
+    result.httpStatus.get match {
+      case 500 => response.internalServerError.json(errorResponse)
+      case 404 => response.notFound.json(errorResponse)
+      case 400 => response.badRequest.json(errorResponse)
+    }
+  }
+
+  private def toError(exception: ResponseException): DisplayError = {
+    // Elasticsearch returns errors as JSON documents, which are stored in the
+    // `message` attribute of ElasticsearchException.  Try to read it as JSON,
+    // so we can check if this was a user error -- but if parsing fails, it's
+    // enough to return a 500 error.
+    try {
+      // Annoyingly, the exact format of message is
+      //
+      //    POST http://localhost:9200/path: HTTP/1.1 500 Internal Server Error
+      //    {"error":{...}}
+      //
+      // so we need to read the second line to get the actual JSON.
+      //
+      // Except when it isn't!  Sometimes these exceptions don't include
+      // a JSON response, so there's only a single-line message -- then `.head`
+      // fails: "NoSuchElementException: next on empty iterator".  I think
+      // this happens when the cluster doesn't reply, but I'm not sure.
+      val jsonDocument = exception.getMessage
+        .split("\n")
+        .tail
+        .head
+      jsonToError(jsonDocument = jsonDocument, exception = exception)
+    } catch {
+      case e: Exception =>
+        serverError(s"Error ($e) parsing Elasticsearch response", exception)
```

Comment:
I don't think these errors should be user facing. We should log the ES error but not pass anything on to the user other than a 500 (we might want to create a request id, so we an match that to the actual error in logs if need be).

Reply:
I agree but this should probably be done in a different PR. This PR should not cause any change in the API behaviour

Reply:
👍